blues_refactor/example.py

```diff
@@ -0,0 +1,85 @@
+"""
+example.py: Provides an example script to run BLUES and
+benchmark the run on a given platform
+
+Authors: Samuel C. Gill
+Contributors: Nathan M. Lim, David L. Mobley
+
+* Benchmarking related code adapted from:
+https://github.com/pandegroup/openmm/blob/master/examples/benchmark.py
+(Authors: Peter Eastmen)
+
+version: 0.0.2 (WIP-Refactor)
+"""
+from __future__ import print_function
+from simtk import unit, openmm
+from simtk.openmm import app
+from alchemy import AbsoluteAlchemicalFactory, AlchemicalState
+
+import blues.utils as utils
+
+import blues.ncmc_switching as ncmc_switching
+from blues.smartdart import SmartDarting
+
+import blues_refactor.ncmc as ncmc
+from blues_refactor.simulationfactory import SimulationFactory
+
+import sys, os, parmed
+import numpy as np
+import mdtraj as md
+from mdtraj.reporters import HDF5Reporter
+from optparse import OptionParser
+
+def runNCMC(platform_name):
+    #Define some options
+    opt = { 'temperature' : 300.0, 'friction' : 1, 'dt' : 0.002,
+            'numIter' : 10, 'nstepsNC' : 10, 'nstepsMD' : 50,
+            'nonbondedMethod' : 'PME', 'nonbondedCutoff': 10, 'constraints': 'HBonds',
+            'trajectory_interval' : 10, 'reporter_interval' : 10, 'platform' : platform_name,
+            'verbose' : True }
+
+    #Defines ncmc move eqns for lambda peturbation of sterics/electrostatics
+    opt['functions'] = { 'lambda_sterics' : 'step(0.199999-lambda) + step(lambda-0.2)*step(0.8-lambda)*abs(lambda-0.5)*1/0.3 + step(lambda-0.800001)',
```

Comment:
@sgill2 - having a dictionary of options can be helpful as it can allow for extension/modification of the options without modifying a list of required arguments. I'm less clear as to whether this specific implementation is the best way to do it or not, but that's the rationale I believe (and to a large extent I buy into it).

@nathanmlim - further comments?